lib/phoenix_ecto/plug.ex

```diff
@@ -23,3 +23,40 @@ unless Ecto.SubQueryError in excluded_exceptions do
     end
   end
 end
+
+unless Phoenix.Ecto.PendingMigrationError in excluded_exceptions do
+  defimpl Plug.Exception, for: Phoenix.Ecto.PendingMigrationError do
+    def status(_error), do: 404
+
+    def actions(%{repo: repo}),
+      do: [
+        %{
+          label: "Run migrations for repo",
+          handler: {__MODULE__, :migrate, [repo]}
+        }
+      ]
+
+    def migrate(repo) do
+      if Process.whereis(repo),
+        do: Ecto.Migrator.run(repo, :up, all: true, dynamic_repo: repo.get_dynamic_repo())
```

Comment:
I don't think we need to pass the dynamic repo because the repo would already be in the dynamic repo.

Also, this is likely to be invoked early enough that a dynamic repo wouldn't be set.

Reply:
[fixed here](https://github.com/phoenixframework/phoenix_ecto/compare/846278a787c3300abfadce9a9d8097da9131afe2..5083d99f7f515143f5d00420d620f5491833597f)